File: rzd.front/src/types/PickUpTrainRequest.ts
export interface PickUpTrainRequest {
  trackedRouteId: number;
  startArrivalTime: string;
  endArrivalTime: string;
  startDepartureTime: string;
  endDepartureTime: string;
  carTypes: string[];
  isUpperSeat: boolean;
  travelTimeInMinutes: number;
  minPrice: number;
  maxPrice: number;
  carServices: string[];
}

File: RZD.Application/Models/PickUpTrainRequest.cs
namespace RZD.Application.Models
{
    public class PickUpTrainRequest
    {
        public long TrackedRouteId { get; set; }
        public DateTime StartArrivalTime { get; set; }

        public DateTime EndArrivalTime { get; set; }

        public DateTime? StartDepartureTime { get; set; }

        public DateTime? EndDepartureTime { get; set; }

        public List<string> CarTypes { get; set; }

        public List<string> CarServices { get; set; }

        public bool IsUpperSeat {get; set;}

        public bool IsLowerSeat { get; set; }
        public bool IsAnySeat { get; set; }

        public int? TravelTimeInMinutes { get; set; }

        public decimal? MinPrice { get; set; }
        public decimal? MaxPrice { get; set; }

    }
}
